ada/extensions/src/libadalang-doc_utils.adb

```diff
@@ -209,9 +209,9 @@ package body Libadalang.Doc_Utils is
    function Get_Documentation (Decl : Basic_Decl) return Doc_Type is
       Doc : Doc_Type;
    begin
-      if Decl.Kind in Ada_Base_Package_Decl
-        and then Decl.Kind /= Ada_Generic_Package_Decl
-      then
+      if Decl.Kind = Ada_Generic_Package_Internal then
+         return Get_Documentation (Decl.Parent.As_Basic_Decl);
+      elsif Decl.Kind in Ada_Base_Package_Decl | Ada_Generic_Package_Decl then
```

Comment:
Looks like this change deserves a ticket, a changes entry and a testcase: could you take care of this?